test/core/test_poll.c

```diff
@@ -9,14 +9,23 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <poll.h>
+#include <emscripten.h>
 
 int main() {
+  EM_ASM(
+    var dummy_device = FS.makedev(64, 0);
+    FS.registerDevice(dummy_device, {});
+
+    FS.createDataFile('/', 'file', 'abcdef', true, true, false);
+    FS.mkdev('/device', dummy_device);
+  );
+  
   struct pollfd multi[5];
-  multi[0].fd = open("/file", O_RDONLY, 0777);
-  multi[1].fd = open("/device", O_RDONLY, 0777);
+  multi[0].fd = open("/file", O_RDWR, 0777);
```

Comment:
Is this due to the poll change?

Reply:
Yeah, it looks like prior to the poll change WasmFS would think none of the flags were set because it was checking if `flags == O_RDONLY` instead of checking that `flags & O_ACCESSMODE == O_RDONLY`, etc.